### ISSUE TITLE
Debugger built-in rule example needs to be fixed

### ISSUE THREAD

**Issue by mchoi8739** [aws/amazon-sagemaker-examples#1286]
### Debugger notebook to be updated:
[The sample notebook of the Debugger built-in rule](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-debugger/tensorflow_builtin_rule/tf-mnist-builtin-rule.ipynb)

### In the **Result** section:
>  As you can see, in the summary, there was no step in the training which reported vanishing gradients in the tensors. Although, the loss was not found to be decreasing at step 1900.

The actual training job does not run further beyond step=1400, so it is not actually showing if the `loss_not_decreasing()` rule detects the issue.

Also the final output graph is too crude. This is because of the default `save_step=500`. 
May want to provide a finer data points for the plot, and need to modify the rule to something like:

```
rules = [
    Rule.sagemaker(
        base_config=rule_configs.loss_not_decreasing(),
        collections_to_save=[ 
            CollectionConfig(
                name="losses", 
                parameters={
                      "train.save_interval": "50"}    
            )
        ]
    )
]
```

**Comment by seanpmorgan** [comment]
Looks to have been fixed in https://github.com/aws/amazon-sagemaker-examples/pull/1866
